rpcs3/rpcs3qt/game_list_frame.cpp

```diff
@@ -580,46 +581,64 @@ void game_list_frame::Refresh(const bool from_drive, const bool scroll_after)
 		{
+			GameInfo game{};
+			game.path = dir_or_elf;
+
 			const Localized thread_localized;
 
-			const std::string sfo_dir = rpcs3::utils::get_sfo_dir_from_game_path(dir);
+			const std::string sfo_dir = rpcs3::utils::get_sfo_dir_from_game_path(dir_or_elf);
 			const psf::registry psf = psf::load_object(sfo_dir + "/PARAM.SFO");
 			const std::string_view title_id = psf::get_string(psf, "TITLE_ID", "");
 
 			if (title_id.empty())
 			{
-				// Do not care about invalid entries
-				return;
+				if (!fs::is_file(dir_or_elf))
+				{
+					// Do not care about invalid entries
+					return;
+				}
+
+				game.serial = dir_or_elf.substr(dir_or_elf.find_last_of(fs::delim) + 1);
+				game.name = game.serial;
+				game.category = "/OS"; // Key for operating system executables
+				game.version = utils::get_firmware_version();
+				game.fw = game.version;
+				game.bootable = 1;
+				game.icon_path = dev_flash + "vsh/resource/explore/icon/icon_home.png";
+			}
+			else
+			{
+				game.serial       = std::string(title_id);
+				game.name         = std::string(psf::get_string(psf, "TITLE", cat_unknown_localized));
+				game.app_ver      = std::string(psf::get_string(psf, "APP_VER", cat_unknown_localized));
+				game.version      = std::string(psf::get_string(psf, "VERSION", cat_unknown_localized));
+				game.category     = std::string(psf::get_string(psf, "CATEGORY", cat_unknown));
+				game.fw           = std::string(psf::get_string(psf, "PS3_SYSTEM_VER", cat_unknown_localized));
+				game.parental_lvl = psf::get_integer(psf, "PARENTAL_LEVEL", 0);
+				game.resolution   = psf::get_integer(psf, "RESOLUTION", 0);
+				game.sound_format = psf::get_integer(psf, "SOUND_FORMAT", 0);
+				game.bootable     = psf::get_integer(psf, "BOOTABLE", 0);
+				game.attr         = psf::get_integer(psf, "ATTRIBUTE", 0);
 			}
 
-			GameInfo game{};
-			game.path         = dir;
-			game.serial       = std::string(title_id);
-			game.name         = std::string(psf::get_string(psf, "TITLE", cat_unknown_localized));
-			game.app_ver      = std::string(psf::get_string(psf, "APP_VER", cat_unknown_localized));
-			game.version      = std::string(psf::get_string(psf, "VERSION", cat_unknown_localized));
-			game.category     = std::string(psf::get_string(psf, "CATEGORY", cat_unknown));
-			game.fw           = std::string(psf::get_string(psf, "PS3_SYSTEM_VER", cat_unknown_localized));
-			game.parental_lvl = psf::get_integer(psf, "PARENTAL_LEVEL", 0);
-			game.resolution   = psf::get_integer(psf, "RESOLUTION", 0);
-			game.sound_format = psf::get_integer(psf, "SOUND_FORMAT", 0);
-			game.bootable     = psf::get_integer(psf, "BOOTABLE", 0);
-			game.attr         = psf::get_integer(psf, "ATTRIBUTE", 0);
-
-			if (m_show_custom_icons)
+			if (m_show_custom_icons && game.icon_path.empty())
```

Comment:
this logic breaks the custom icon for vsh.

Reply:
Fixed.